Despair-Games/poketernity
# [Refactor] Decouple form change from evo phase, reduce nesting in form-change-phase.ts

## What are the changes the user will see?
none


## Why am I making these changes?
- evo phase and form change phase are entangled together
- callback hell on form change phase



## What are the changes from a developer perspective?
- vanquished the callback hell in `form-change-phase.ts`
- added abstract class `FormChangeBasePhase` for `FormChangePhase` and `EvolutionPhase`
- changed `EvolutionSceneHandler` to `FormChangeSceneHandler` to be more generic and not attached to evo
- renamed `Mode.EVOLUTION_SCENE` to `Mode.FORM_CHANGE_SCENE`, same reasoning as above


## Screenshots/Videos
<img width="403" alt="Screenshot 2024-12-14 at 2 01 52 AM" src="https://github.com/user-attachments/assets/75deec01-5967-48f4-8fb2-78220ef32aa9" />




## How to test the changes?
- `npm run test` (see `form-change-phase.test.ts`)
- play up to a pokemon's evolution
- do a form change to a pokemon ie Ice Face Eiscue to Noice Eiscue and vice versa



## Checklist


Are there any localization additions or changes? If so:
